src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.LocalFunctionReferenceRewriter.cs

```diff
@@ -82,10 +82,8 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
                         out method,
                         ref arguments);
 
-                    var newType = _lambdaRewriter.VisitType(node.Type);
```

Comment:
@AlekseyTs And I talked about this offline.

It turns out to be slightly harder than I expected to assert all the things I want to assert.

Rather than spend a lot of time to only get a little extra value here, I've filed https://github.com/dotnet/roslyn/issues/16057 to try to do more of these things in a structured way.